subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformationSubject.java

```diff
@@ -183,8 +182,8 @@ public SubsequentTransformationSubject(TransformationSubject previous, Immutable
         }
 
         @Override
-        public ArtifactTransformDependencies getDependencies() {
-            return previous.getDependencies();
+        public ArtifactTransformDependenciesProvider getArtifactDependenciesProvider() {
+            return previous.getArtifactDependenciesProvider();
         }
```

Comment:
Would it make sense to pass in the provider in some other path than the subject? Like, directly passing it in some method parameter where it needs to be used..?

Reply:
The alternative is adding an extra parameter to `Transform.transform` and `TransformerInvoker.invoke`. Both are internal APIs, so that's not an issue. Both approach make sense, I'll go with the other one in the follow up PR and we can decide to keep the change or drop it.